### ISSUE TITLE
False positive for RpC_REPEATED_CONDITIONAL_TEST

### ISSUE THREAD

**Issue by ksilz** [spotbugs/spotbugs#1768]
The rule is [RpC_REPEATED_CONDITIONAL_TEST](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#rpc-repeated-conditional-tests-rpc-repeated-conditional-test). This `if` condition triggers it - but there the two conditional tests are different:
```java
if (visitDay.equals(bookingDay) == false || (visitDay.equals(bookingDay) && visitTimeOfDay != bookingTimeOfDay))
```
